### PR TITLE
Modify MetronomeAlarmThread for moving into OMR

### PR DESCRIPTION
**Pull request by jason-hall**
Signed-off-by: Jason Hall <jasonhal@ca.ibm.com>

### CODE REVIEW

File: runtime/gc_realtime/MetronomeAlarmThread.cpp [hunk 1]
```diff
@@ -163,13 +136,12 @@ MM_MetronomeAlarmThread::metronomeAlarmThreadWrapper(void* userData)
...
 {
-	J9JavaVM *vm = (J9JavaVM *)env->getLanguageVM();
-	UDATA startPriority;
+	uintptr_t startPriority;
 	bool retCode;
 
 	startPriority = J9THREAD_PRIORITY_MAX;
 
-	if (0 != vm->internalVMFunctions->createThreadWithCategory(
+	if (0 != createThreadWithCategory(
```
Comment (dmitripivkine) [inline runtime/gc_realtime/MetronomeAlarmThread.cpp]: Would be better to use J9THREAD_SUCCESS instead of zero. This is definition from OMR

**Comment by dmitripivkine** [comment]
Jenkins test sanity xLinux jdk8

**Comment by dmitripivkine** [comment]
Failed:
```
14:24:54 not ok 91 - J9vmTest_2
14:24:54   ---
14:24:54     output:
14:24:54       |
14:24:54         ===============================================
14:24:54         Running test J9vmTest_2 ...
14:24:54         ===============================================
14:24:54         J9vmTest_2 Start Time: Mon Feb 11 14:56:00 2019 Epoch Time (ms): 1549911360235
14:24:54         variation: Mode351
14:24:54         JVM_OPTIONS: -Xcompressedrefs -Xgcpolicy:metronome -Xcompressedrefs 
14:24:54         Unhandled exception
14:24:54         Type=Segmentation error vmState=0xffffffff
14:24:54         J9Generic_Signal_Number=00000004 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000080
14:24:54         Handler1=00007FF6C2133940 Handler2=00007FF6C151E330 InaccessibleAddress=0000000000000000
14:24:54         RDI=00007FF6C2891300 RSI=00007FF6C0012C50 RAX=207883481474C085 RBX=00007FF6BC4DF9A8
14:24:54         RCX=00007FF6C34C42C0 RDX=00007FF6BBD87549 R8=00007FF6C0012B80 R9=00007FF6BC4DBA90
14:24:54         R10=0000000000000000 R11=0000000000000246 R12=0000000000000001 R13=0000000000000000
14:24:54         R14=00007FF6BBBA6110 R15=00007FF6BC4DFE18
14:24:54         RIP=00007FF6BBC93C56 GS=0000 FS=0000 RSP=00007FF6C0012C50
14:24:54         EFlags=0000000000010246 CS=0033 RBP=00007FF6C2891300 ERR=0000000000000000
14:24:54         TRAPNO=000000000000000D OLDMASK=0000000000000000 CR2=00007F8758798700
14:24:54         xmm0 0000000000000000 (f: 0.000000, d: 0.000000e+00)
14:24:54         xmm1 0000000000000000 (f: 0.000000, d: 0.000000e+00)
14:24:54         xmm2 00ff000000000000 (f: 0.000000, d: 7.063274e-304)
14:24:54         xmm3 00000000000000ff (f: 255.000000, d: 1.259867e-321)
14:24:54         xmm4 0000000000000000 (f: 0.000000, d: 0.000000e+00)
14:24:54         xmm5 0000000000000000 (f: 0.000000, d: 0.000000e+00)
14:24:54         xmm6 2e6b636f4c246e77 (f: 1277456000.000000, d: 4.405733e-85)
14:24:54         xmm7 0000000000000000 (f: 0.000000, d: 0.000000e+00)
14:24:54         xmm8 705f7366632e7570 (f: 1663989120.000000, d: 1.953116e+233)
14:24:54         xmm9 0000000000000000 (f: 0.000000, d: 0.000000e+00)
14:24:54         xmm10 0000000000000000 (f: 0.000000, d: 0.000000e+00)
14:24:54         xmm11 ffffffffffffffff (f: 4294967296.000000, d: -nan)
14:24:54         xmm12 0000000000000000 (f: 0.000000, d: 0.000000e+00)
14:24:54         xmm13 0000000000000000 (f: 0.000000, d: 0.000000e+00)
14:24:54         xmm14 0000000000000000 (f: 0.000000, d: 0.000000e+00)
14:24:54         xmm15 0000000000000000 (f: 0.000000, d: 0.000000e+00)
14:24:54         Module=/home/jenkins/workspace/PullRequest-Sanity-JDK8-linux_x86-64_cmprssptrs-OpenJ9/build/linux-x86_64-normal-server-release/images/j2sdk-image/jre/lib/amd64/compressedrefs/libj9gc29.so
14:24:54         Module_base_address=00007FF6BBB24000
14:24:54         Target=2_90_20190211_675 (Linux 4.4.0-140-generic)
14:24:54         CPU=amd64 (8 logical CPUs) (0x3eae99000 RAM)
14:24:54         JVMDUMP039I Processing dump event "gpf", detail "" at 2019/02/11 14:56:00 - please wait.
14:24:54         JVMDUMP032I JVM requested System dump using '/home/jenkins/workspace/PullRequest-Sanity-JDK8-linux_x86-64_cmprssptrs-OpenJ9/openj9/test/TestConfig/test_output_1549909558318/J9vmTest_2/core.20190211.145600.13909.0001.dmp' in response to an event
14:24:54         JVMPORT030W /proc/sys/kernel/core_pattern setting "|/usr/share/apport/apport %p %s %c %d %P" specifies that the core dump is to be piped to an external program.  Attempting to rename either core or core.13922.
14:24:54         
14:24:54         JVMDUMP010I System dump written to /home/jenkins/workspace/PullRequest-Sanity-JDK8-linux_x86-64_cmprssptrs-OpenJ9/openj9/test/TestConfig/test_output_1549909558318/J9vmTest_2/core.20190211.145600.13909.0001.dmp
14:24:54         JVMDUMP032I JVM requested Java dump using '/home/jenkins/workspace/PullRequest-Sanity-JDK8-linux_x86-64_cmprssptrs-OpenJ9/openj9/test/TestConfig/test_output_1549909558318/J9vmTest_2/javacore.20190211.145600.13909.0002.txt' in response to an event
14:24:54         JVMDUMP010I Java dump written to /home/jenkins/workspace/PullRequest-Sanity-JDK8-linux_x86-64_cmprssptrs-OpenJ9/openj9/test/TestConfig/test_output_1549909558318/J9vmTest_2/javacore.20190211.145600.13909.0002.txt
14:24:54         JVMDUMP032I JVM requested Snap dump using '/home/jenkins/workspace/PullRequest-Sanity-JDK8-linux_x86-64_cmprssptrs-OpenJ9/openj9/test/TestConfig/test_output_1549909558318/J9vmTest_2/Snap.20190211.145600.13909.0003.trc' in response to an event
14:24:54         JVMDUMP010I Snap dump written to /home/jenkins/workspace/PullRequest-Sanity-JDK8-linux_x86-64_cmprssptrs-OpenJ9/openj9/test/TestConfig/test_output_1549909558318/J9vmTest_2/Snap.20190211.145600.13909.0003.trc
14:24:54         JVMDUMP013I Processed dump event "gpf", detail "".
14:24:54         
14:24:54         Unhandled exception in signal handler. Protected function: reportThreadCrash (0x0)
14:24:54         
14:24:54         
14:24:54         J9vmTest_2_FAILED
14:24:54     duration_ms: 5614
```

**Comment by dmitripivkine** [comment]
also `cmdLineTester_jvmtitests_5` crashed similar way

**Comment by jason-hall** [comment]
createThreadWithCategory() in vm/threadhelp.cpp contains identical code to createThreadWithCategory() in omr/util/omrutil/threadhelp.c. I don't know why both exist.

**Comment by jason-hall** [comment]
The parameters to signalProtectedFunction() were incorrect for use as a function pointer argument to j9sig_protect(). It should be fixed shortly.

**Comment by dmitripivkine** [comment]
Jenkins test sanity xLinux jdk8

**Comment by dmitripivkine** [comment]
Jenkins test extended Aix jdk8

**Pull request merged**